R/read_from_file_helpers.R

```diff
@@ -0,0 +1,281 @@
+#' Get file base name without its extension
+#'
+#' @param x the file path
+#'
+#' @return a `character` with the file base name(s) without it extension
+#'
+#' @examples
+#' \dontrun{
+#' base_name <- get_base_name(
+#'   x = system.file("extdata", "test.txt", package = "readepi")
+#' )
+#' }
+#'
+get_base_name <- function(x) {
+  ext <- rio::get_ext(x)
+  bn <- gsub(paste0(".", ext), "", basename(x))
+  bn
+}
+
+#' Detect separator from a string
+#'
+#' @param x a string
+#'
+#' @return a `character vector` of identified separators
+#'
+#' @examples
+#' \dontrun{
+#' sep <- detect_separator(
+#'   x = "My name is Karim"
+#' )
+#' }
+#'
+detect_separator <- function(x) {
+  special_characters <- c("\t", "|", ",", ";", " ", ":")
+  sep <- vector(mode = "character", length = 0)
+  for (spec_char in special_characters) {
+    if (grepl(spec_char, x)) {
+      sep <- c(sep, spec_char)
+    }
+  }
+  unique(sep)
```

Comment:
some of the characters like the space character could be found several times in a string. the `unique()` function will serve in resolving this by preventing from sending out multiple space characters.

Reply:
I don't think that's correct. Since you loop over `special_characters`, `sep` will necessarily be a subset of `special_characters`, without duplicates. You can verify this by removing `unique()` and running the example from L29

Reply:
That is actually correct. I will remove the `unique()` function.